pkgs/misc/moonfire-nvr/default.nix

```diff
@@ -0,0 +1,73 @@
+{ lib
+, rustPlatform
+, buildNpmPackage
+, fetchFromGitHub
+, pkg-config
+, ncurses
+, sqlite
+, symlinkJoin
+, testers
+}:
+
+let
+  pname = "moonfire-nvr";
+  version = "0.7.6";
+  src = fetchFromGitHub {
+    owner = "scottlamb";
+    repo = "moonfire-nvr";
+    rev = "v${version}";
+    hash = "sha256-hPgS4Y/dD6G8lqfsJz3aeeed6P+ngJpBOng88xUc55Q=";
+  };
+
+  server = rustPlatform.buildRustPackage {
+    inherit version src;
+    pname = "${pname}-server";
+    sourceRoot = "source/server";
+    cargoLock = {
+      lockFile = ./Cargo.lock;
+      outputHashes = {
+        "hashlink-0.8.1" = "sha256-h7DEapTVy0SSTaOV9rCkdH3em4A9+PS0k1QQh1+0P4c=";
+        "mp4-0.9.2" = "sha256-mJZJDzD8Ep9c+4QusyBtRoqAArHK9SLdFxG1AR7JydE=";
+      };
+    };
+    nativeBuildInputs = [
+      pkg-config
+    ];
+    buildInputs = [
+      ncurses
+      sqlite
+    ];
+    doCheck = false;
+  };
+  ui = buildNpmPackage {
+    inherit version src;
+    pname = "${pname}-ui";
+    sourceRoot = "source/ui";
+    npmDepsHash = "sha256-IpZWgMo6Y3vRn9h495ifMB3tQxobLeTLC0xXS1vrKLA=";
+    postInstall = "cp -r build $out/lib/ui";
+  };
+in symlinkJoin {
```

Comment:
It would be better to just copy the ui into the server build to have a more standard build